src/main/java/dk/kb/image/TemplateMostUsedColors.java

```diff
@@ -0,0 +1,136 @@
+            for (int x = 0; x < width; x++) {
+                // Get RGB for each pixel
+                pixelCount ++;
+                int pixelRGB = img.getRGB(x, y);
+                updateBucketCounter(pixelRGB, buckets, bucketCounter);
+            }
+        }
+        return bucketCounter;
+    }
+
+    /**
+     * Method to update bucketCounter in getBucketCount().
+     * @param pixel The current pixels color value in RGB colorspace.
+     * @param buckets List of color buckets of used type in the implementation.
+     * @param bucketCounter Integer array to store count of buckets.
+     */
+    void updateBucketCounter(int pixel, List<C> buckets, int[] bucketCounter) {
+        // Convert RGB pixel value to OK Lab float
+        // Values for checking max
+        int bestColor = 0;
+        double minDistance = 2147483647;
+        for (int i = 0; i < buckets.size(); i ++){
+            double totalDistance = calculateDistance(pixel, buckets.get(i));
+            // Evaluates total distance against minimum distance for given bucket
+            if (totalDistance < minDistance) {
+                minDistance = totalDistance;
+                bestColor = i;
+            }
+        }
+        // Add 1 to the bucket closest to pixel color
+        bucketCounter[bestColor] ++;
+    }
+
+    abstract double calculateDistance(int pixel, C bucket);
+
+    /**
+     * Combine list of color-buckets and bucketCount int[] into a map. The list holds the keys and the int[] holds the values.
+     * The arrays get combined by index number.
+     * @param buckets list of implemented type with keys of the map.
+     * @param bucketCount int[] with the values for the map.
+     * @return a map with key-value pairs from the input list and int[].
+     */
+     protected Map<C, Integer> combineBucketsAndBucketCount(List<C> buckets, int[] bucketCount){
+         Map<C, Integer> bucketsWithCount = new HashMap<>();
+         for (int i=0; i<buckets.size(); i++) {
+             bucketsWithCount.put(buckets.get(i), bucketCount[i]);
+         }
+         return bucketsWithCount;
+    }
+
+    /**
+     * Method to sort a map<C, Integer> and have the key-value pair with the biggest value at index 0.
+     * @param bucketsWithCount of type map<C, Integer> to be sorted
+     * @return a list of entries of the type Entry<C, Integer> sorted after biggest value.
+     */
+    List<Map.Entry<C, Integer>> sortList(Map<C, Integer> bucketsWithCount){
+        List<Map.Entry<C, Integer>> sortedList = new ArrayList<>(bucketsWithCount.entrySet());
+        sortedList.sort(Map.Entry.comparingByValue());
+        Collections.reverse(sortedList);
+
+        return sortedList;
+    }
+
+    /**
+     * Return a JSON array with top X entries from input list.
+     * @param sortedList input list to extract top x from.
+     * @param x integer to limit size of returned list.
+     * @return a JSON array containing the first x entries from the input list.
+     */
```

Comment:
What are the "entries"? (RGB colors and their percentage)
What does the input list hold? (palette colors and their percentage)